index.js

```diff
@@ -44,10 +44,12 @@ module.exports = function configParser(yaml, templateFactory) {
         .then(phaseGeneratePermutations)
         // Output in the right format
         .then(doc => ({
+            annotations: Hoek.reach(doc, 'annotations', { default: {} }),
```

Comment:
Ahh got it. Thanks for the explanations @d2lam and @minz1027 .

I think we should leave this code as-is for now, and add in the `models` in a future PR to `data-schema` when relevant. As @minz1027 said, this won't break anything. I'll add a note to the original issue (https://github.com/screwdriver-cd/screwdriver/issues/580) indicating that we still need to modify the`pipeline` database.